Vungle/src/main/java/com/applovin/mediation/adapters/VungleMediationAdapter.java

```diff
@@ -403,19 +407,65 @@ private void updateUserPrivacySettings(final MaxAdapterParameters parameters)
         }
     }
 
-    private static BannerAdSize vungleAdSize(final MaxAdFormat adFormat)
+    private static int getApplicationWindowWidth(final Context context)
     {
-        if ( adFormat == MaxAdFormat.BANNER )
+        WindowManager windowManager = (WindowManager) context.getSystemService( Context.WINDOW_SERVICE );
+        Display display = windowManager.getDefaultDisplay();
+        DisplayMetrics outMetrics = new DisplayMetrics();
+        display.getMetrics( outMetrics );
+        return outMetrics.widthPixels;
+    }
+
+    private int getAdaptiveBannerWidth(final MaxAdapterParameters parameters, final Context context)
+    {
+        if ( AppLovinSdk.VERSION_CODE >= 11_00_00_00 )
         {
-            return BannerAdSize.BANNER;
+            final Map<String, Object> localExtraParameters = parameters.getLocalExtraParameters();
+            Object widthObj = localExtraParameters.get( "adaptive_banner_width" );
+            if ( widthObj instanceof Integer )
+            {
+                return (int) widthObj;
+            }
+            else if ( widthObj != null )
+            {
+                log( "Expected parameter \"adaptive_banner_width\" to be of type Integer, received: " + widthObj.getClass() );
+            }
         }
-        else if ( adFormat == MaxAdFormat.LEADER )
-        {
-            return BannerAdSize.BANNER_LEADERBOARD;
+
+        int deviceWidthPx = getApplicationWindowWidth( context );
+        return AppLovinSdkUtils.pxToDp( context, deviceWidthPx );
+    }
+
+    private VungleAdSize vungleAdSize(
```

Comment:
Do we support custom size here?

Reply:
No, because MAX guide not pass fixed width and height to us.

Reply:
iOS side, we can pass custom size with code below. Can you do similar for Android side? 

```
// Custom Banner Size request
CGFloat height = 300;
CGFloat width = 400;
self.adView.frame = CGRectMake(0, 0, width, height);
[self.adView setExtraParameterForKey: @"adaptive_banner" value: @"false"];
[self.adView setLocalExtraParameterForKey: @"adaptive_banner_width" value: [[NSNumber numberWithFloat:width] stringValue]];
[self.adView setLocalExtraParameterForKey: @"adaptive_banner_height" value: [[NSNumber numberWithFloat:height] stringValue]];
```

Reply:
And, this is not on their integration guide tho.